src/components/wishes/AddToWish.tsx

```diff
@@ -0,0 +1,50 @@
+import useCheckwished from '../../hooks/useCheckWishlist';
+import { FaHeart, FaRegHeart } from 'react-icons/fa';
+import { createWish } from '../../redux/features/createWishesSlice';
+import { useAppDispatch } from '../../redux/hooks/hooks';
+import { DynamicData } from '../../@types/DynamicData';
+import useToast from '../../hooks/useToast';
+import { fetchWishes } from '../../redux/features/getUserwishes';
+type AddToWishType = {
+	productId: string;
+	btnCss?: string;
+};
+function AddToWish({ productId, btnCss }: AddToWishType) {
+	const { wished, setWished } = useCheckwished(productId);
+	const dispatch = useAppDispatch();
+	const { showErrorMessage, showSuccessMessage } = useToast();
+
+	const handleAddRemoveWish = async () => {
+		try {
+			await dispatch(createWish({ productId })).unwrap();
+			await dispatch(fetchWishes()).unwrap();
+			if (wished) {
+				showSuccessMessage('removed to wish successfully');
+				setWished(!wished);
+			} else {
+				showSuccessMessage('Added to wish successfully');
+				setWished(!wished);
```

Comment:
here's as well the response message should be coming from backend. not hard coded.